tiny_cnn/core/kernels/avx_conv2d_back_kernel.h

```diff
@@ -0,0 +1,626 @@
+                        dst3 = madd(w3a, delta_src, dst3);
+                        dst4 = madd(w4a, delta_src, dst4);
+                        _mm256_storeu_ps(delta_dst0, dst0);
+                        _mm256_storeu_ps(delta_dst1, dst1);
+                        _mm256_storeu_ps(delta_dst2, dst2);
+                        _mm256_storeu_ps(delta_dst3, dst3);
+                        _mm256_storeu_ps(delta_dst4, dst4);
+                        delta_dst0 += w_stride;
+                        delta_dst1 += w_stride;
+                        delta_dst2 += w_stride;
+                        delta_dst3 += w_stride;
+                        delta_dst4 += w_stride;
+                    } // for x
+                    pdelta_src += out.width_;
+                    pdelta_dst += h_stride2;
+                } // for y
+            } // for outc
+        } // for inc
+    }
+
+    // accumulate dw
+    if (out.width_ == 1 && out.height_ == 1) {
+        const float* pprev_out = &prev_out[0];
+        for (size_t inc=0; inc<in.depth_; ++inc, pprev_out+=in_padded_area) {
+            VECTORIZE_ALIGN(32) float floats[28];
+            size_t in_padded_width = in_padded.width_;
+            _mm256_store_ps(&floats[0], _mm256_loadu_ps(pprev_out + in_padded_width * 0));
+            _mm256_storeu_ps(&floats[5], _mm256_loadu_ps(pprev_out + in_padded_width * 1));
+            _mm256_storeu_ps(&floats[10], _mm256_loadu_ps(pprev_out + in_padded_width * 2));
+            _mm256_storeu_ps(&floats[15], _mm256_loadu_ps(pprev_out + in_padded_width * 3));
+            _mm256_storeu_ps(&floats[20], _mm256_loadu_ps(pprev_out + in_padded_width * 4));
+            __m256 prevos0 = _mm256_load_ps(&floats[0]);
+            __m256 prevos1 = _mm256_load_ps(&floats[8]);
+            __m256 prevos2 = _mm256_load_ps(&floats[16]);
+            __m128 prevos3 = _mm_load_ss(&floats[24]);
+            cnn_size_t widx = 25 * inc;
+            cnn_size_t widx_delta = 25 * in.depth_;
+            float* pdW = &dW[widx];
+            for (cnn_size_t outc = 0; outc < out.depth_; outc++, pdW+=widx_delta) {
+                if (!tbl.is_connected(outc, inc)) {
+                    continue;
+                }
+                __m256 delta = _mm256_broadcast_ss(&curr_delta[outc]);
+                __m256 w0 = _mm256_loadu_ps(pdW+0);
+                __m256 w1 = _mm256_loadu_ps(pdW+8);
+                __m256 w2 = _mm256_loadu_ps(pdW+16);
+                __m128 w3 = _mm_load_ss(pdW+24);
+                w0 = madd(prevos0, delta, w0);
+                w1 = madd(prevos1, delta, w1);
+                w2 = madd(prevos2, delta, w2);
+                w3 = madd_ss(prevos3, _mm256_castps256_ps128(delta), w3);
+                _mm256_storeu_ps(pdW+0, w0);
+                _mm256_storeu_ps(pdW+8, w1);
+                _mm256_storeu_ps(pdW+16, w2);
+                _mm_store_ss(pdW+24, w3);
+            }
+        }
+    }else {
+        // prepare load-mask beforehand
```

Comment:
I'm not sure how I should reformat this code. Can you give me some instructions?